float32 rotate_radius
float32 back_mid_vel